### PR TITLE
Adding spiceinit classic as back up to Ale implementation. 

### PR DESCRIPTION
**Pull request by Kelvinrr**
Updated Spice::init to default back to the old ISIS implementation in case of failure. Still needs an update to ALE to work. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Spice/Spice.cpp [hunk 1]
```diff
@@ -170,41 +170,54 @@ namespace Isis {
...
+        load(kernels["LeapSecond"], noTables);
+        if ( kernels.hasKeyword("SpacecraftClock")) {
+          load(kernels["SpacecraftClock"], noTables);
+        }
+        m_usingAle = true;
+      } 
+      catch(std::invalid_argument &e) {
+         // Backup to stadnard ISIS implementation
+         m_usingAle = false; 
+         if (noTables) {
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.cpp]: Indent is off here

File: isis/src/base/objs/Spice/Spice.cpp [hunk 2]
```diff
@@ -170,41 +170,54 @@ namespace Isis {
...
+        // try using ALE
+        std::ostringstream kernel_pvl;
+        kernel_pvl << kernels;
+
+        json props;
+        props["kernels"] = kernel_pvl.str();
+
+        isd = json::parse(ale::loads(cube.fileName().toStdString(), props.dump(), "isis"));
+        json aleNaifKeywords = isd["NaifKeywords"];
+        m_naifKeywords = new PvlObject("NaifKeywords", aleNaifKeywords);
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.cpp:184]: Does this expect the NaifKeywords from ALE to be a PVL formatted string embedded in the JSON object?

File: isis/src/base/objs/Spice/Spice.cpp [hunk 3]
```diff
@@ -170,41 +170,54 @@ namespace Isis {
...
+        
+        // Still need to load clock kernels for now
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.cpp:186]: Any reason why this is the case?

File: isis/src/base/objs/Spice/Spice.cpp [hunk 4]
```diff
@@ -1157,7 +1170,7 @@ namespace Isis {
     // Read from PvlObject that is our naif keywords
     QVariant result;
 
-    if (m_naifKeywords->hasKeyword(key) && !m_usingNaif) {
+    if (m_naifKeywords->hasKeyword(key) && m_usingAle) {
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.cpp]: Why is this changed? We still want to be able to read from the NaifKeywords object even when we didn't use ALE.

File: isis/src/base/objs/Spice/Spice.h [hunk 5]
```diff
@@ -435,6 +435,8 @@ namespace Isis {
 
       bool m_usingNaif; /**< Indicates whether we are reading values from the 
                              NaifKeywords PvlObject in cube*/
+
+      bool m_usingAle;
```
Comment (jessemapel) [inline isis/src/base/objs/Spice/Spice.h]: Needs docs

File: isis/src/base/objs/Spice/Spice.cpp (hunk 3 above)
Comment (Kelvinrr) [reply isis/src/base/objs/Spice/Spice.cpp:186]: Has to do with the weird way times are added to naif keywords.

in Spice.cpp:
```
 1006   iTime Spice::getClockTime(QString clockValue, int sclkCode, bool clockTicks) {
 1007     if (sclkCode == -1) {
 1008       sclkCode = naifSclkCode();
 1009     }
 1010
 1011     iTime result;
 1012
 1013     QString key = "CLOCK_ET_" + Isis::toString(sclkCode) + "_" + clockValue;
 1014     QVariant storedClockTime = getStoredResult(key, SpiceDoubleType);
```

Ale isn't shipping the information to construct this string. The only way it seems to add this functionality without changing all Camera Plugins is to pack this info into the ISD.

File: isis/src/base/objs/Spice/Spice.cpp (hunk 2 above)
Comment (Kelvinrr) [reply isis/src/base/objs/Spice/Spice.cpp:184]: no, PvlObject has a constructor that converts the json format to PVL. So the JSON is the standard json object in ALE and the returned PVL is formatted exactly as ISIS expects it to be.

File: isis/src/base/objs/Spice/Spice.cpp (hunk 4 above)
Comment (Kelvinrr) [reply isis/src/base/objs/Spice/Spice.cpp]: changed it to be more explicit.

**Review by jessemapel — APPROVED** [review]
